proposals/0000-provenance-qualified-imports.rst

```diff
@@ -0,0 +1,107 @@
+
+One use for an overlay may be an internal company repo. To take a concrete example, suppose ``foocorp`` has an internal repo, provided within their intranet as an overlay. In that overlay is a package: ``common-network-utils``. Then, sometime later, somebody uploads a totally different ``common-network-utils`` to hackage. Now, it may be that their build resolves to the wrong package! By providing provenance-qualification on package imports, this situation is prevented. One might say, "wouldn't specifying carefully the precedence of overlays and root repos fix this? Well, it would, if it was never the case that both packages (which share nothing but a name!) are never depended on at once. But it may be that some upstream dependency incurs a dependency on ``hackage+common-network-utils``. By extending povenance qualification throughout namespacing, this proposal makes it possible to depend on both the ``foocorp-internal+common-network-utils`` package *and* the ``hackage+common-network-utils`` package, without any worry of overlap. 
+
+Additionally, this proposal allows depdencies on packages hosted outside of a repo or overlay -- such as github, just on the web more generally, or within a filesystem (such as on a shared drive). A use case here would be if I had a package undergoing a major rewrite, and which I did not yet want to release a new version of, but which I wanted to encourage others to take for a test-drive. Rather than asking users to install it directly, or clone and vendor it into their tree, I could now tell them how to use provenance qualification to depend on a particular snapshot of the source repo through altering a few lines in their cabal file.
+
+Proposed Change Specification
+-----------------------------
+
+A) *Provenance Information*
+
+A new optional field, `source-origin` is added to the `InstalledPackageInfo <http://downloads.haskell.org/~ghc/latest/docs/html/users_guide/packages.html#installed-pkg-info>`_ file and in turn to ghc package databases. Tools which register packages to databases may also specify this field. The intended use of this field is to contain a url  that specifies the origin of the package, typically the root of the package repository from which a package was sourced.
+
+B) *Provenance Aliases*
+
+Since urls can be unwieldy, and it is expected that few will be used, it makes sense to allow specification of shorthand aliases to ghc with a new flag. Hence we add a new flag `--source-origin-alias [ALIAS]=[URL]``  that may be specified multiple times. In the event the same alias is specified twice, an error is thrown.
+
+C) *Provenance Specification*
+
+With ``-XPackageImports`` we can specify e.g.
+
+``import "somepackage" Foo.Bar``
+
+Under this proposal we can also specify
+
+``import "[REPO]+somepackage" Foo.Bar``
+
+Where repo is either the url of a repository root or an alias which maps to one.
+
+D) Cabal syntax
```

Comment:
I don't believe that Cabal syntax changes belong to the scope of a proposal. So I suggest moving this section out of the *Specification* section, and move it to the *Effect and interactions* section, as an example and justification that the proposal meets its goal.